src/main/scala/strawman/collection/immutable/NumericRange.scala

```diff
@@ -80,6 +82,8 @@ final class NumericRange[T](
     else locationAfterN(idx)
   }
 
+  def updated[B >: T](index: Int, elem: B): IndexedSeq[B] = collection.generic.Updated.toImmutableArray(index, elem, coll)
```

Comment:
The `Range` is not materialized here, only the resulting `IndexedSeq`. I would expect this operation to behave with a strict semantics since `Range` is not a lazy collection type.